server/src/main/java/org/opensearch/monitor/fs/FsProbe.java

```diff
@@ -123,6 +142,12 @@ final FsInfo.IoStats ioStats(final Set<Tuple<Integer, Integer>> devicesNumbers,
                     final long sectorsRead = Long.parseLong(fields[5]);
                     final long writesCompleted = Long.parseLong(fields[7]);
                     final long sectorsWritten = Long.parseLong(fields[9]);
+                    // readTime and writeTime calculates the total read/write time taken for each request to complete
+                    // ioTime calculates actual time queue and disks are busy
+                    final long readTime = Long.parseLong(fields[6]);
+                    final long writeTime = Long.parseLong(fields[10]);
+                    final long ioTime = fields.length > 12 ? Long.parseLong(fields[12]) : 0;
```

Comment:
Why do we have explicit check if length is greater than 12 ? is it possible the `proc/diskstats` file won't have this ioTime for some linux  machines. 

If that is the case making it 0 is not giving correct information right. ioTime is 0 indicates IO was idle it can possible normally also so we cannot differentiate it.  or can we make as  -1 instead of 0.

let me know if I missing anything here.

Reply:
-1 represents previous data missing in the context of fsstats. So not using -1 here. 

These fields are almost defintely present but I can only trace back to linux kernel versions 2.14 - so in order to not receive indexoutofbounds in edge cases , chose to go with this approach.

We will still be able to differentiate that kernel doesn't have these fields based on other parameters such as operations, throughput - if they are non empty and these fields are 0.